ext/plotting.jl

```diff
@@ -85,6 +85,121 @@ function Attractors.heatmap_basins_attractors!(ax, grid, basins, attractors;
     return ax
 end
 
+function colors_from_keys_shaded(ukeys)
+    # Unfortunately, until `to_color` works with `Cycled`,
+    # we need to explicitly add here some default colors...
+    COLORS =[ 
+        :black,
+        :red,
+        :green, 
+        :blue,
+        :purple,
+        :yellow,]
+    n = length(COLORS)
+    v_col = []
+    vals = zeros(2*length(ukeys))
+    for k in eachindex(ukeys)
+        push!(v_col, COLORS[mod(k-1,n)+1])
+    end
+
+    return Dict(k => v_col[i] for (i, k) in enumerate(ukeys))
+end
+
+function custom_colormap_shaded(ukeys)
+    # Light and corresponding dark colors for shading of basins of attraction
+    LIGHT_COLORS = [
+        :gray95,
+        :lightsalmon,
+        :darkseagreen1,
+        :azure,
+        :lavenderblush,
+        :lightyellow,]      
+    DARK_COLORS = [
+        :black,
+        :red4,
+        :darkgreen,
+        :navyblue,
+        :purple4,
+        :gold4,]      
+    n = length(LIGHT_COLORS)
```

Comment:
This would make the same colors if there are more than 6 attractors, right?

Reply:
The colors are circled mod 6. So no problem on this side. Anyway I will stick to the defined colors and adjust the brightness for each one.